android/BOINC/app/src/main/java/edu/berkeley/boinc/BOINCActivity.kt

```diff
@@ -203,11 +220,17 @@ class BOINCActivity : AppCompatActivity() {
         }
         super.onPause()
         unregisterReceiver(mClientStatusChangeRec)
+//        unregisterReceiver(mRpcExternBroadCastReceiver)
     }
 
     private fun doBindService() {
         // start service to allow setForeground later on...
-        startService(Intent(this, Monitor::class.java))
+        if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
```

Comment:
[`ContextCompat.startForegroundService()`](https://developer.android.com/reference/kotlin/androidx/core/content/ContextCompat#startforegroundservice) can be used here.

Reply:
Maybe but that version isn't really supporting it and might cause all kind of problems.

Reply:
The same thing is done other parts of the code.
I don't think this is needed in older versions. That said there are many changes in Services to keep it in the Foreground and it's still up to the phone to keep it running or kill it.